RevenueCatUI/Templates/V2/PaywallsV2View.swift

```diff
@@ -0,0 +1,286 @@
+//
+//  File.swift
+//
+//
+//  Created by Josh Holtz on 6/11/24.
+//
+// swiftlint:disable missing_docs
+
+import RevenueCat
+import SwiftUI
+
+#if PAYWALL_COMPONENTS
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+private class PaywallStateManager: ObservableObject {
+    @Published var state: Result<PaywallState, Error>
+
+    init(state: Result<PaywallState, Error>) {
+        self.state = state
+    }
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+private struct PaywallState {
+
+    let viewModelFactory: ViewModelFactory
+    let packages: [Package]
+    let componentViewModel: PaywallComponentViewModel
+    let showZeroDecimalPlacePrices: Bool
+
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct PaywallsV2View: View {
```

Comment:
Some related comments around naming: 
- The use of `V2` in the name. It's not terrible, but also not very descriptive. 
- "Paywalls" being plural is a bit confusing. It might seem that this is going to show multiple paywalls at once, but it's not.
- If I understand correctly, this is at the same level architecturally as `LoadedOfferingPaywallView`. Would be good to name it consistently to indicate that.

Maybe these comments can be resolved by naming changes such as: 
- `LoadedOfferingPaywallView` -> `LoadedLegacyPaywallView` or `LoadedTemplatePaywallView`
- `PaywallsV2View` -> `LoadedPaywallView`. If we move the creation of `PaywallState` one level up, we may not need the existing `LoadedPaywallsV2View`?